### PR TITLE
New component scaffolded: <nylas-scheduler>

### PR DESCRIPTION
**Pull request by philrenaud**
Currently just wraps `<nylas-availability>` and has a listener for `timeSlotChosen`
CleanShot 2021-08-11 at 15 25 08

### CODE REVIEW

**Review by wonka-lighthouse — COMMENTED** [review]
| Device | URL |
|--|--|
| mobile | http://localhost:8000/components/agenda/src/index.html |

     
| Device | URL |
|--|--|
| mobile | http://localhost:8000/components/composer/src/index.html |

Not what you expected? Are your scores flaky? [Run Lighthouse on Foo](https://www.foo.software/lighthouse)
If scores continue to be inconsistent consider [running all audits on Foo](https://github.com/foo-software/lighthouse-check-action#usage-foos-automated-lighthouse-check-api)

**Review by arjunkalburgi — APPROVED** [review]
g2g

File: components/scheduler/README.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+# Nylas scheduler
+
+Nylas scheduler (`<nylas-contact-list>`) is part of the Nylas Components library that lets you build user-facing contact functionality into your application in minutes. Use Nylas scheduler with your Nylas account or by passing in your own JSON data.
```
Comment (heathergh) [inline components/scheduler/README.md:3]: `Nylas scheduler (`<nylas-contact-list>`)`? Should this be `(`nylas-scheduler`)`?

File: components/scheduler/README.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+![Nylas scheduler example](https://nylas-static-assets.s3-us-west-2.amazonaws.com/public-documentation/contact_list_gif.gif)
```
Comment (heathergh) [inline components/scheduler/README.md:5]: This also references `contact_list`

File: components/scheduler/README.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+- npm `npm i @nylas/components-contact-list`
+- Yarn `yarn add npm i @nylas/components-contact-list`
+
+Alternatively, on an html page, you can load the scheduler using a script tag:
+
+```
+   <head>
+      <!-- Import the script from CDN -->
+      <script src="https://unpkg.com/@nylas/components-contact-list"></script>
```
Comment (heathergh) [inline components/scheduler/README.md:35]: contact-list again

File: components/scheduler/README.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    </head>
+```
+
+For both installation options, in the body of your page, you can instantiate the scheduler with `<nylas-contact-list></nylas-contact-list>`.
```
Comment (heathergh) [inline components/scheduler/README.md:39]: get out of here, contact-list!

File: components/scheduler/README.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+1. [Fetching data directly from Nylas](#fetching-data-directly-from-nylas)
+2. [Passing in your own data](#passing-in-your-own-data)
+
+### Fetching data directly from Nylas
+
+#### Setup
+
+If you haven't registered for a Nylas account yet, you can do so at dashboard.nylas.com. Once there, head to the **Components** tab and create a new scheduler component.
+
+You'll be guided through the component setup and be given the option to tie your component to your calendar account. Review the [scheduler Documentation](https://developer.nylas.com/docs/user-experience/components/contact-list-component/).
```
Comment (heathergh) [inline components/scheduler/README.md:54]: ✂️

File: components/scheduler/README.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+#### Allowed domains
+
+During the setup process, you'll be prompted to provide a list of [allowed domains](https://developer.nylas.com/docs/user-experience/components/contact-list-component/#allowed-domains). Be sure to add any domains you'll be testing your app on, including `localhost`, and any staging and production URLs you might use.
```
Comment (heathergh) [inline components/scheduler/README.md:58]: ✂️

File: components/scheduler/README.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+### Passing in your own data
+
+Nylas scheduler can be used as a UI on top of any contacts that you provide. Events should follow the [Nylas contacts object standard](https://developer.nylas.com/docs/api/#tag--Contacts--contact-object).
```
Comment (heathergh) [inline components/scheduler/README.md:62]: ✂️

File: components/scheduler/README.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+      ],
+      given_name: "Albert Einstein"
+    }
+  ];
+```
+
+Then pass the array into your component using any JavaScript.
+
+```js
+<nylas-contact-list contacts={staticContacts}>
```
Comment (heathergh) [inline components/scheduler/README.md:98]: ✂️

File: components/scheduler/README.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+```
+
+You can also use plain JavaScript as an attribute.
+
+```js
+document.querySelector("nylas-contact-list").contacts = staticContacts;
```
Comment (heathergh) [inline components/scheduler/README.md:104]: ✂️

File: components/scheduler/README.md [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+```
+
+### Properties
+
+Nylas scheduler allows for several properties that affect the layout and functionality of your component. You can find a complete list of properties within our [Documentation for Nylas scheduler](https://developer.nylas.com/docs/user-experience/components/contact-list-component/#customization)
```
Comment (heathergh) [inline components/scheduler/README.md:109]: ✂️

File: components/scheduler/README.md [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+For example, you can listen for a `contactClicked` event with the following code:
+
+```js
+document
+  .querySelector("nylas-contact-list")
+  .addEventListener("contactClicked", (event) => {
+    let { detail } = event;
+    console.log("contact clicked", detail);
+  });
+```
```
Comment (heathergh) [inline components/scheduler/README.md:124]: ✂️

File: components/scheduler/README.md [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+### Testing
+
+`yarn cy:open` will launch our end-to-end tests in a browser
+tests will automatically be run on push from push.yaml
+snapshot (visual) diff tests are fun using Percy by running `yarn snapshot`
+
+## Additional Documentation
+
+- [Nylas scheduler on NPM](https://www.npmjs.com/package/@nylas/components-contact-list)
+- [Nylas Docs: scheduler Component](https://developer.nylas.com/docs/user-experience/components/contact-list-component/)
```
Comment (heathergh) [inline components/scheduler/README.md:143]: ✂️

**Review by heathergh — APPROVED** [review]
LGTM - just need to change all the references of `contact-list` to `scheduler`

**Pull request merged**